commit 7cd972a85705341dd8306eefc558ed9e5def05d7
Author: Maarten Lankhorst <maarten.lankhorst@canonical.com>
Date:   Thu Feb 20 11:14:03 2014 +0100

    return immediately in preinit when called with PROBE_DETECT
    
    This fixes a crash with Xorg -configure.
    
    Bug:
    https://bugs.launchpad.net/ubuntu/+source/xorg/+bug/1278046
    
    Signed-off-by: Maarten Lankhorst <maarten.lankhorst@canonical.com>
    Reviewed-by: Alex Deucher <alexander.deucher@amd.com>

diff --git a/src/radeon_kms.c b/src/radeon_kms.c
index 21a420f..4a6c38e 100644
--- a/src/radeon_kms.c
+++ b/src/radeon_kms.c
@@ -781,6 +781,9 @@ Bool RADEONPreInit_KMS(ScrnInfoPtr pScrn, int flags)
     uint32_t tiling = 0;
     int cpp;
 
+    if (flags & PROBE_DETECT)
+        return TRUE;
+
     xf86DrvMsgVerb(pScrn->scrnIndex, X_INFO, RADEON_LOGLEVEL_DEBUG,
 		   "RADEONPreInit_KMS\n");
     if (pScrn->numEntities != 1) return FALSE;
